bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/connector/PowerMaxReaderThread.java

```diff
@@ -0,0 +1,182 @@
+                            // Postamble
+
+                            if (tmpData[i] != 0x0A && dataBuffer[index - 1] == 0x43) {
+                                // adjust message length for 0x43
+                                msgLen++;
+                            } else if (checkCRC(dataBuffer, index)) {
+                                // whole message received with a right CRC
+
+                                byte[] msg = new byte[index];
+                                msg = Arrays.copyOf(dataBuffer, index);
+
+                                connector.setWaitingForResponse(System.currentTimeMillis());
+                                connector.handleIncomingMessage(msg);
+
+                                // find new preamble
+                                index = 0;
+                            } else if (msgLen == 0) {
+                                // CRC check failed for a message with an
+                                // unknown length
+                                logger.debug("Message length is now {} but message is apparently not complete",
+                                        index + 1);
+                                logData = Arrays.copyOf(dataBuffer, index);
+                                logger.debug("Partial message {}", DatatypeConverter.printHexBinary(logData));
+                            } else {
+                                // CRC check failed for a message with a known
+                                // lenght
+                                logData = Arrays.copyOf(dataBuffer, index);
+                                logger.debug("Message with CRC error {}", DatatypeConverter.printHexBinary(logData));
+
+                                connector.setWaitingForResponse(System.currentTimeMillis());
+
+                                // find new preamble
+                                index = 0;
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (InterruptedIOException e) {
+            Thread.currentThread().interrupt();
+            logger.debug("Interrupted via InterruptedIOException");
+        } catch (IOException e) {
+            logger.debug("Reading failed: {}", e.getMessage());
+        }
+
+        logger.debug("Data listener stopped");
+    }
+
+    /**
+     * Check if the CRC inside a received message is valid or not
+     *
+     * @param data
+     *            the buffer containing the message
+     * @param len
+     *            the size of the message in the buffer
+     *
+     * @return true if the CRC is valid or false if not
+     */
+    private boolean checkCRC(byte[] data, int len) {
+        byte checksum = PowerMaxCommDriver.computeCRC(data, len);
+        byte expected = data[len - 2];
+        if (checksum != expected) {
+            logger.debug("CRC check failed (expected {}, got {})", String.format("%02X", expected),
+                    String.format("%02X", checksum));
```

Comment:
Yes, I changed to WARN level and I added the display of the message content in this log entry.